test/parallel/test-trace-events-perf.js

```diff
@@ -51,37 +51,36 @@ if (process.argv[2] === 'child') {
     const file = path.join(tmpdir.path, 'node_trace.1.log');
 
     assert(fs.existsSync(file));
-    fs.readFile(file, common.mustCall((err, data) => {
-      const traces = JSON.parse(data.toString()).traceEvents
-        .filter((trace) => trace.cat !== '__metadata');
-      assert.strictEqual(traces.length,
-                         expectedMarks.length +
-                         expectedBegins.length +
-                         expectedEnds.length);
+    const data = fs.readFileSync(file);
```

Comment:
I don't get it, why `fs.readFileSync()` instead of `fs.readFile()`? The callback of `fs.readFile()` is indeed only called once with the entire contents of the file no?

Reply:
I agree with @lpinca. from the docs:
> ```
>  The fs.readFile() function buffers the entire file. To minimize memory costs, when possible prefer 
>  streaming via fs.createReadStream().
>  ```
It might be that the forked process does not flush its file write buffer before signaling `exit`?

Change looks good, but I'm not sure it solves the problem, or it solves it in a non obvious way.

Reply:
Ooooh, my bad....